### ISSUE TITLE
`klint` adds a line break after block assignment regardless of `max_line_length`

### ISSUE THREAD

**Issue by x80486** [pinterest/ktlint#2464]
## Expected Behavior

Block assignment should not break if it fits on its own line, based on `max_line_length`:

```kotlin
configure<GitPropertiesPluginExtension> {
  keys = buildList {
    add("git.branch")
    add("git.commit.id")
  }
}
```

## Observed Behavior

`klint` adds a line break right after the assignment:

```kotlin
configure<GitPropertiesPluginExtension> {
  keys =
    buildList {
      add("git.branch")
      add("git.commit.id")
    }
}
```

> **NOTE:** I'm using the Spotless plugin in a Gradle project. I haven't found how to configure `--log-level`, but I've raised an issue in the Spotless repository to see if that could be added.

## Steps to Reproduce

Have Spotless (or probably just `klint`) installed and configured. In my case, I'm not using any additional configuration. In a Gradle build script paste this:

```kotlin
val keys = buildList {
  add("git.branch")
  add("git.commit.id")
}
```

And then run `./gradlew :spotlessApply`

## Environment

My `.editorconfig` looks like:

```properties
root = true

[*]
continuation_indent_size = 4
indent_size = 2
indent_style = space
max_line_length = 120

# Keep these unchanged
charset = utf-8
end_of_line = lf
insert_final_newline = true
trim_trailing_whitespace = true
```

Spotless is configured this way:

```kotlin
configure<SpotlessExtension> {
  encoding = StandardCharsets.UTF_8
  kotlinGradle {
    endWithNewline()
    indentWithSpaces(2)
    ktlint("1.1.0")
    trimTrailingWhitespace()
  }
  lineEndings = LineEnding.UNIX
}
```

I'm using Gradle `8.5` under Linux, and `klint` is set at version `1.1.0`.

**Comment by paul-dingemans** [comment]
This behavior is caused by the `multiline-expression` rule. You can [disable](https://pinterest.github.io/ktlint/latest/faq/#how-do-i-enable-or-disable-a-rule) it in your `.editorconfig`.

**Comment by x80486** [comment]
Tried to locate `multiline-expression` but couldn't find it — the closest one is [multiline-expression-wrapping](https://pinterest.github.io/ktlint/latest/rules/standard/#multiline-expression-wrapping).

Provided that's the correct one, I went ahead and placed this in the end of the `.editorconfig` file:

```properties
...

[*.{kt,kts}]
ktlint_standard_multiline-expression-wrapping = disabled
```

But still, it's doing exactly the same :grimacing:

In any case, if you don't think this shouldn't be the default behavior, feel free to close this issue. I would think that wrapping will make the most sense when lines are too long based on the defined `max_line_length` — with the few exceptions of some constructs that are always wrapped, like `enum` values, class annotations, etc.

Thanks for maintaining `klint`! :wave:

**Comment by paul-dingemans** [comment]
> Tried to locate `multiline-expression` but couldn't find it — the closest one is [multiline-expression-wrapping](https://pinterest.github.io/ktlint/latest/rules/standard/#multiline-expression-wrapping).

Yes, this is correct.


> NOTE: I'm using the Spotless plugin in a Gradle project. I haven't found how to configure --log-level, but I've raised an issue in the Spotless repository to see if that could be added.

This could be a spotless problem. I don't know how they integrate with Ktlint. But with ktlint CLI it works, but you need to disable another rule as well because of rule dependencies:
```
root = true

[*.{kt,kts}]
ktlint_standard_multiline-expression-wrapping = disabled
ktlint_standard_string-template-indent = disabled
```

**Comment by x80486** [comment]
It works when the new rule, `ktlint_standard_string-template-indent`, is disabled in `.editorconfig` as well.

Thanks, @paul-dingemans!
